### PR TITLE
Advanced/Ingress: Review configuration & add WAF.

### PR DESCRIPTION
**Pull request by Gacko**
### What does this PR do?

This PR reviews the advanced Ingress configuration documentation and adds information about how to enable & configure the Web Application Firewall capabilities of the Ingress NGINX Controller.

### What does it look like?

Textual change only.

### Any background context you can provide?

We were asked for the Web Application Firewall capabilities of the Ingress NGINX Controller.

(Closes https://github.com/giantswarm/giantswarm/issues/25911)

### What is needed from the reviewers?

You should be able to install & configure the Ingress NGINX Controller in a workload cluster so you can check if the added documentation is easily understandable and working.

### Have you maintained the front matter?

Yes, I did. I bumped the last review date and added user questions regarding the Web Application Firewall part.

### CODE REVIEW

File: src/content/advanced/ingress/configuration/index.md [hunk 1]
```diff
@@ -443,11 +431,12 @@ metadata:
...
   values: |
-    configmap:
-      log-format-upstream: "MY EDITED LOG FORMAT - $status $body_bytes_sent $http_referer"
+    controller:
+      config:
+        log-format-upstream: "MY EDITED LOG FORMAT - $status $body_bytes_sent $http_referer"
 ```
 
-However keep in mind that with great power comes great responsibility.
+However keep in mind: With great power comes great responsibility! 🕷️
```
Comment (AndiDog) [inline src/content/advanced/ingress/configuration/index.md:439]: Not sure if customers know what the bug depicts at this specific location 😉

**Review by AndiDog — COMMENTED** [review]
LGTM but there's an insect

File: src/content/advanced/ingress/configuration/index.md (hunk 1 above)
Comment (Gacko) [reply src/content/advanced/ingress/configuration/index.md:439]: Yeah, you're probably right. I was already thinking about linking the Wiki page (there actually is one about this quote), but gonna remove the emoji.

**Review by AndiDog — APPROVED** [review]

**Comment by Gacko** [comment]
Closed https://github.com/giantswarm/giantswarm/issues/25911.

File: src/content/advanced/ingress/configuration/index.md [hunk 2]
```diff
@@ -379,23 +371,22 @@ apiVersion: networking.k8s.io/v1
 kind: Ingress
 metadata:
   name: INGRESS_NAME
-  namespace: NAMESPACE
```
Comment (ced0ps) [inline src/content/advanced/ingress/configuration/index.md:382]: Not sure why you'd remove the namespace?

File: src/content/advanced/ingress/configuration/index.md [hunk 3]
```diff
@@ -515,6 +508,135 @@ data:
...
+          #SecAuditLogFormat JSON
+
+          # Disable rule 920350 due to false-positives on health checks.
+          SecRuleRemoveById 920350
+
+        # Optional: Enable OWASP ModSecurity Core Rule Set.
+        enable-owasp-modsecurity-crs: "true"
+```
+
+The above configuration snippet makes ModSecurity send its logs to `/dev/stdout`, so you can process them in the same way you're doing for access & error logs of Ingress NGINX Controller itself. Additionally it disables [rule 920350](https://github.com/coreruleset/coreruleset/blob/v3.3/master/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf#L708-L736), which blocks direct access by IP, which is required for the Kubernetes probes to work. Without disabling this rule, your Ingress NGINX Controller containers would start to repeatingly get killed the moment you activate the blocking mode. Apart from that, you would see a lot of spam regarding this false-positive in your logs even in detection mode.
```
Comment (ced0ps) [inline src/content/advanced/ingress/configuration/index.md:563]: "repeatedly"

**Review by ced0ps — COMMENTED** [review]
LGTM

**Pull request merged**